DeepSpeech.py

```diff
@@ -481,51 +445,51 @@ def train():
                 sys.exit(1)
 
         # Retrieving global_step from restored model and setting training parameters accordingly
-        model_feeder.set_data_set(no_dropout_feed_dict, train_set)
-        step = session.run(global_step, feed_dict=no_dropout_feed_dict)
+        step = session.run(global_step)
         num_gpus = len(Config.available_devices)
-        steps_per_epoch = max(1, train_set.total_batches // num_gpus)
-        steps_trained = step % steps_per_epoch
```

Comment:
How do you offset steps of a half-done epoch now? Same question on the feeding side.

Reply:
You don't. For a lot of the experiments we've been doing it doesn't make any sense to try and start from the middle of an epoch, since there's no guarantee that the previous training run used the same datasets as the current one. My plan is to follow up on this PR with another one that makes all training runs start from epoch 0, and make the global step be the "version of a checkpoint", which is both more precise and matches the language used by other researchers. Then we can say "release 0.5.0 was trained on this data for 75000 steps" in releases, and there's no confusion of epoch counts varying wildly when the dataset changes.

Reply:
See for example:

https://discourse.mozilla.org/t/increasing-train-batch-size-2-to-train-batch-size-3-causes-deepspeech-not-to-train-anymore-why/29790
https://discourse.mozilla.org/t/what-does-the-test-of-epoch-number-mean/29770
https://discourse.mozilla.org/t/problem-in-using-checkpoints-different-datasets-start-with-different-epoch-value/37407
https://discourse.mozilla.org/t/checkpoint-resuming-no-of-epochs-do-not-match-on-google-colab/37358

Reply:
A long time ago we already discussed about it, I remember now. Bold move. I also like "release 0.5.0 was trained on this data for 75000 steps".